### DISCUSSION TITLE
Aliases for long filenames/urls?

### DISCUSSION THREAD

**Discussion by thedatadoc** [chdb-io/chdb#129] [Q&A]
I'm new to chdb and am a big fan.  Great work!  I'm curious - is there a way for me to create an alias for a long filename or url so that my users can simplify their queries?  I'd prefer to avoid text substitution as that is surely the path to ruin.  Their SQL statements will never be as simple as I imagine.

**Comment by lmangani** [comment]
Hello @thedatadoc and thanks for the kind words! For persistent sessions, you should be able to achieve this using SET ClickHouse parameters to provide arbitrary variables into any query. Here's an example I use for [urleng](https://get.urleng.com/) purely as a reference:

```
:) SET param_url = 'https://some.webserver.com/abc123'
:) SELECT * FROM url({url:String}, JSONEachRow)
```

Is this compatible with your goal?

**Reply by thedatadoc** [reply]
Thanks!  Your answer and @auxten's are pretty close.  I was hoping for something as simple as a plain table name that would fit the user's mental model of how they'd typically work with other database systems, but I appreciate the suggestion.

**Comment by auxten** [comment]
In Python, you can use something like this:
```python
urlPath = "https://some.webserver.com/abc123"
SQL = f"SELECT * FROM url({urlPath}, JSONEachRow)"
```
